catalog/app/containers/Bucket/PackageDialog/PackageDialog.js

```diff
@@ -287,35 +380,62 @@ export function MetaInput({ schemaError, input, meta, schema }) {
         </Lab.ToggleButtonGroup>
       </div>
 
-      {value.mode === 'kv' ? (
-        <JsonEditor
-          error={error}
-          disabled={disabled}
-          value={parsedValue}
-          onChange={onJsonEditor}
-          schema={schema}
-        />
-      ) : (
-        <M.TextField
-          variant="outlined"
-          size="small"
-          value={value.text}
-          onChange={handleTextChange}
-          placeholder="Enter JSON metadata if necessary"
-          error={!!error}
-          helperText={
-            !!error &&
-            {
-              jsonObject: 'Metadata must be a valid JSON object',
-            }[error]
-          }
-          fullWidth
-          multiline
-          rowsMax={10}
-          InputProps={{ classes: { input: classes.jsonInput } }}
-          disabled={disabled}
-        />
-      )}
+      <div {...getRootProps({ className: classes.dropzone })} tabIndex={undefined}>
+        {value.mode === 'kv' ? (
+          <JsonEditor
+            error={error}
+            disabled={disabled}
+            value={parsedValue}
+            onChange={onJsonEditor}
+            schema={schema}
+            key={jsonEditorKey}
+          />
+        ) : (
+          <M.TextField
+            variant="outlined"
+            size="small"
+            value={value.text}
+            onChange={handleTextChange}
+            placeholder="Enter JSON metadata if necessary"
+            error={!!error}
+            helperText={
+              !!error &&
+              {
+                jsonObject: 'Metadata must be a valid JSON object',
+                schema: 'Metadata must conform to the schema',
+              }[error]
+            }
+            fullWidth
+            multiline
+            rowsMax={10}
+            InputProps={{ classes: { input: classes.jsonInput } }}
+            disabled={disabled}
+          />
+        )}
+
+        {(isDragActive || locked) && (
```

Comment:
For me, it's simpler to read when splitting this into two different blocks:

```
{isDragActive && (
  <div overlay>
    Drop file
  </div>
)}

{(locked && !isDragActive) && (
  <CircularProgress />
)}
```

---

Also, mixing two styles:

* isDragActive, isLocked
* dragActive, locked

Reply:
> For me, it's simpler to read when splitting this into two different blocks:

this way we have to repeat the wrapper part

> Also, mixing two styles

`isDragActive` comes from the external API, so i have no control over it (well, ofc i can rename it, but i dont think that's necessary)